test/language/class/class-strict-mode-with.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class strict mode, with
+flags: [noStrict]
+negative: SyntaxError
+---*/
+with ({a: 1}) {
+  assert.sameValue(a, 1);
+}
+
+class C extends (function B() { with ({}); return B; }()) {}
```

Comment:
with is not allowed in strict mode.

http://people.mozilla.org/~jorendorff/es6-draft.html#sec-strict-mode-code

 * All parts of a ClassDeclaration or a ClassExpression are strict mode code.

@bterlson If the test does not have a [noStrict] does it run the test in strict and sloppy mode? If so, then the noStrict flag can be removed here too.